llvm/lib/Target/AMDGPU/AMDGPUSplitModule.cpp

```diff
@@ -358,22 +358,24 @@ struct KernelWithDependencies {
                          const DenseMap<const Function *, CostType> &FnCosts,
                          const Function *Fn)
       : Fn(Fn) {
-    addAllDependencies(SML, CG, *Fn, Dependencies, HasIndirectCall,
-                       HasExternallyVisibleDependecy);
+    addAllDependencies(SML, CG, *Fn, Dependencies, HasIndirectCall);
     TotalCost = FnCosts.at(Fn);
     for (const auto *Dep : Dependencies) {
       TotalCost += FnCosts.at(Dep);
-      HasExternallyVisibleDependecy |= Dep->hasExternalLinkage();
+
+      // We cannot duplicate functions with external linkage, or functions that
+      // may be overriden at runtime.
+      HasNonDuplicatableDependecy |=
+          (Dep->hasExternalLinkage() || !Dep->isDefinitionExact());
```

Comment:
Maybe this should be mayBeDerefined? I'm not a linkage expert, but it seems there's some nobuiltin interaction here too

Reply:
`mayBeDerefined` is private I think, `isDefinitionExact` is just `!mayBeDerefined` so it's already using it